modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts

```diff
@@ -102,4 +125,240 @@ export default class Ecdsa {
+      Ecdsa.curve.scalarMult(BigInt(yShare.j), BigInt(xShare.i)),
+      BigInt(xShare['x']),
+      Ecdsa.curve.scalarInvert(d),
+    ].reduce(Ecdsa.curve.scalarMult);
+
+    const signers: SignShareRT = {
+      wShare: {
+        i: xShare.i,
+        l: xShare.l,
+        m: xShare.m,
+        n: xShare.n,
+        y: xShare.y,
+        k: bigIntToHex(k),
+        w: bigIntToHex(w),
+        gamma: bigIntToHex(gamma),
+      },
+      kShare: {} as KShare,
+    };
+
+    signers.kShare = {
+      i: yShare.j,
+      j: xShare.i,
+      n: bigIntToHex(pk.n),
+      k: bigIntToHex(pk.encrypt(k)),
+    };
+
+    return signers;
+  }
+
+  /**
+   * Perform multiplicative-to-additive (MtA) share conversion with another
+   * signer by skipping computation of wShare and directly computing
+   * beta and alpha share (Short cut method for 2:n signing threshold)
+   * @param {SignConvert}
+   * @returns {SignConvertRT}
+   */
+  signConvert(shares: SignConvert): SignConvertRT {
+    let shareParticipant: BShare | GShare, shareToBeSend: AShare | MUShare;
+    let isGammaShare = false;
+    if (shares.xShare && shares.yShare && shares.kShare) {
+      const xShare = shares.xShare; // currentParticipant secret xShare
+      const yShare = shares.yShare;
+      const signShare = this.signShare(xShare, yShare);
+      shareToBeSend = { ...shares.kShare, alpha: '', mu: '' } as AShare;
+      shareParticipant = { ...signShare.wShare, beta: '', nu: '' } as BShare;
+    } else if ((shares.bShare && shares.muShare) || (shares.aShare && shares.wShare)) {
+      isGammaShare = true;
+      shareToBeSend = shares.aShare ? ({ ...shares.aShare } as MUShare) : ({ ...shares.muShare } as MUShare);
+      shareParticipant = shares.wShare ? ({ ...shares.wShare } as GShare) : ({ ...shares.bShare } as GShare);
+    } else {
+      throw new Error('Invalid config for Sign Convert');
+    }
+    if (shareParticipant.i !== shareToBeSend.i) {
+      throw new Error('Shares from same participant');
+    }
+    if (shareToBeSend['alpha']) {
+      const pk = getPaillierPublicKey(BigInt(shareParticipant.n));
+      const sk = new paillierBigint.PrivateKey(
+        BigInt(shareParticipant.l as string),
+        BigInt(shareParticipant.m as string),
+        pk
+      );
+      const alpha = sk.decrypt(BigInt(shareToBeSend.alpha));
+      shareParticipant['alpha'] = bigIntToHex(Ecdsa.curve.scalarReduce(alpha));
+      const mu = sk.decrypt(BigInt(shareToBeSend.mu as string)); // recheck encrypted number
+      shareParticipant['mu'] = bigIntToHex(Ecdsa.curve.scalarReduce(mu));
+      delete shareParticipant['l'];
+      delete shareParticipant['m'];
+      delete shareToBeSend['alpha'];
+      delete shareToBeSend['mu'];
+    }
+    if (shareToBeSend['k']) {
+      const n = BigInt(shareToBeSend['n']); // Pallier pub from other signer
+      let pk = getPaillierPublicKey(n);
+      const k = BigInt(shareToBeSend['k']);
+
+      const beta0 = Ecdsa.curve.scalarRandom();
```

Comment:
beta0 and nu0, the original range was to from 1 to n (Where n is paillier constant for its public key), But the scalar negation for noble only permits the operation within the secp256k1 curve range, so had to stick with scalar random, @johnoliverdriscoll hope its okay

Reply:
I've looked over another implementation and it seems to use a beta value that is within secp256k1's curve order, so I'm sure it's fine this way.

Reply:
Ok I had a review of the white paper and I have found that this value (and n0) needs to be within 1 <= beta0 < n. To negate it on the curve you'll simply have to call curve.scalarReduce on it first.